components/experimental/src/units/converter_factory.rs

```diff
@@ -168,11 +180,11 @@ impl ConverterFactory {
         ///     For example, `newton` has the basic units:  `gram`, `meter`, and `second` (each one has it is own power and si prefix).
         fn insert_non_basic_units(
             factory: &ConverterFactory,
-            units: &[SingleUnit],
+            units: &SingleUnitVec,
```

Comment:
suggestion: pass a `&[SingleUnit]` and abstract away the representation
```suggestion
            units: &[SingleUnit],
```